pennylane/resource/resource.py

```diff
@@ -20,6 +20,7 @@
 from abc import abstractmethod
```

Comment:
Not really relevant to this PR. But I realized we can do some bizarre stuff with the Resources object, e.g. using imaginary numbers of wires, strings instead of numbers, and numbers of gates that don't match with the gate_types or gate_sizes. Not a problem, but have we thought about this? 
```
>>> Resources(num_wires=13j, num_gates=1e112, gate_types={'H':1, 'S':0, 'made_up':-12, 'error': 'b'}, gate_sizes={'some_string': 'a lot',1:12,2:3, 'z':12},depth = 12)
wires: 13j
gates: 1e+112
depth: 12
shots: Shots(total=None)
gate_types:
{'H': 1, 'S': 0, 'made_up': -12, 'error': b}
gate_sizes:
{some_string: a lot, 1: 12, 2: 3, z: 12}

Reply:
Could be useful to have some warnings set up that check this kind of thing.